components/content_settings/core/browser/brave_content_settings_pref_provider.h

```diff
@@ -35,10 +40,30 @@ class BravePrefProvider : public PrefProvider {
       ContentSettingsType content_type,
       const ResourceIdentifier& resource_identifier,
       std::unique_ptr<base::Value>&& value) override;
+  std::unique_ptr<RuleIterator> GetRuleIterator(
+      ContentSettingsType content_type,
+      const ResourceIdentifier& resource_identifier,
+      bool incognito) const override;
+
+ private:
+  void UpdateCookieRules(ContentSettingsType content_type, bool incognito);
+  void OnCookieSettingsChanged(ContentSettingsType content_type);
+
+  // content_settings::Observer overrides:
+  void OnContentSettingChanged(const ContentSettingsPattern& primary_pattern,
+                               const ContentSettingsPattern& secondary_pattern,
+                               ContentSettingsType content_type,
+                               const std::string& resource_identifier) override;
+  void NotifyChanges(const std::vector<Rule>& rules, const bool incognito);
 
   // PrefProvider::pref_change_registrar_ alreay has plugin type.
   PrefChangeRegistrar brave_pref_change_registrar_;
 
+  std::map<bool, std::vector<Rule>> cookie_rules_;
```

Comment:
I'd prefer to have a comment describing this mapping from boolean into something